seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSource.java

```diff
@@ -239,44 +204,47 @@ public void setJobContext(JobContext jobContext) {
         this.jobContext = jobContext;
     }
 
-    private void setDeserialization(Config config) {
-        if (config.hasPath(SCHEMA.key())) {
-            Config schema = config.getConfig(SCHEMA.key());
-            // todo: use KafkaDataTypeConvertor here?
-            typeInfo = CatalogTableUtil.buildWithConfig(config).getSeaTunnelRowType();
-            MessageFormat format = ReadonlyConfig.fromConfig(config).get(FORMAT);
-            switch (format) {
-                case JSON:
-                    deserializationSchema = new JsonDeserializationSchema(false, false, typeInfo);
-                    break;
-                case TEXT:
-                    String delimiter = DEFAULT_FIELD_DELIMITER;
-                    if (config.hasPath(FIELD_DELIMITER.key())) {
-                        delimiter = config.getString(FIELD_DELIMITER.key());
-                    }
-                    deserializationSchema =
-                            TextDeserializationSchema.builder()
-                                    .seaTunnelRowType(typeInfo)
-                                    .delimiter(delimiter)
-                                    .build();
-                    break;
-                case CANAL_JSON:
-                    deserializationSchema =
-                            CanalJsonDeserializationSchema.builder(typeInfo)
-                                    .setIgnoreParseErrors(true)
-                                    .build();
-                    break;
-                default:
-                    throw new SeaTunnelJsonFormatException(
-                            CommonErrorCode.UNSUPPORTED_DATA_TYPE, "Unsupported format: " + format);
-            }
-        } else {
-            typeInfo = CatalogTableUtil.buildSimpleTextSchema();
-            this.deserializationSchema =
-                    TextDeserializationSchema.builder()
-                            .seaTunnelRowType(typeInfo)
-                            .delimiter(TextFormatConstant.PLACEHOLDER)
+    private void setDeserialization(
+            ReadonlyConfig option, SeaTunnelDataType<SeaTunnelRow> dataType) {
+        deserializationSchema = getDeserializationSchema(option, dataType);
+    }
+
+    private DeserializationSchema<SeaTunnelRow> getDeserializationSchema(
+            ReadonlyConfig option, SeaTunnelDataType<SeaTunnelRow> typeInfo) {
+        MessageFormat format = option.get(FORMAT);
+        switch (format) {
+            case JSON:
+                if (typeInfo instanceof SeaTunnelRowType) {
```

Comment:
If SeaTunnelRowType is needed, why not update the method to `private DeserializationSchema<SeaTunnelRow> getDeserializationSchema(
            ReadonlyConfig option, SeaTunnelRowType typeInfo)` ?